mlir/lib/Dialect/Rock/Tuning/RockTuningImpl.cpp

```diff
@@ -477,7 +477,8 @@ TuningParamSet *createTunableParamSpace(ModuleOp mod, TuningParamSetKind kind) {
     return WalkResult::interrupt();
   });
   if (!findPrimary.wasInterrupted() && !findAttention.wasInterrupted()) {
-    delete newSpace;
+    llvm_unreachable(
```

Comment:
If we don't find a matching op, newSpace is deleted and then returned, causing additional obscure barfs.  A proposed improvement is to change the interface so this function can return nullptr and the caller can detect and handle the situation.

Reply:
This is `emitFatalError()` or such more than unreachable